stream.go

```diff
@@ -0,0 +1,510 @@
+package molecule
+
+import (
+	"bytes"
+	"io"
+	"math"
+
+	"google.golang.org/protobuf/encoding/protowire"
+)
+
+const (
+	// wire types; see https://developers.google.com/protocol-buffers/docs/encoding
+	wtVarint          int = 0
+	wt64Bit           int = 1
+	wtLengthDelimited int = 2
+	wt32Bit           int = 5
+
+	// defaultBufferSize is the default size for buffers used for embedded
+	// values, which must first be written to a buffer to determine their
+	// length.  This is not used if BufferFactory is set.
+	defaultBufferSize int = 1024 * 16
```

Comment:
This feels pretty big to me. Any reason you selected it? I would have guessed 1kib was a good starting point, but I don't care too much since its configable

Reply:
I had no particular reason for this, other than to avoid the first 15 or so realloc-and-copy operations.  I think a page is a good size for a buffer, and that's almost always 4k.  Looking around the Go source, it seems to prefer 8k in many cases.  Let's compromise there? :)